imperial_coldfront_plugin/views.py

```diff
@@ -36,11 +37,73 @@
     check_group_owner_or_superuser,
     user_already_has_hpc_access,
     user_eligible_for_hpc_access,
+    user_eligible_to_be_pi,
 )
 
 User = get_user_model()
 
 
+@login_required
+def research_group_terms_view(request: HttpRequest) -> HttpResponse:
+    """View for accepting T&Cs and creating a ResearchGroup.
+
+    Args:
+        request: The Http request including the user information.
+
+    Returns:
+        The relevant Http response, depending on the permissions and the type of
+        request.
+    """
+    graph_client = get_graph_api_client()
+    user_profile = graph_client.user_profile(request.user.username)
+    if not user_eligible_to_be_pi(user_profile) and not request.user.is_superuser:
```

Comment:
Ohhhh.... do we want superusers to be able to use this view? Hmm... that's tricky. Probably not I guess. It would allow superusers to potentially make their own individual groups which is probably just messy.

Reply:
True, but otherwise I don't know how to actually test the interface, as I am not an eligible PI. Not sure if I can force that with a local user, but I cannot use them until https://github.com/ImperialCollegeLondon/imperial_coldfront_plugin/pull/154 is merged.

Reply:
Maybe allow superusers to do this ONLY if running in debug mode?

Reply:
I've left access to the superusers and maybe we can go back to this at a later time?

Reply:
Sure.